### PR TITLE
Support jwt authentication on sn-platform

### PR DESCRIPTION
**Pull request by tuteng**
*(If this PR fixes a github issue, please add `Fixes #<xyz>`.)*

Fixes #<xyz>

*(or if this PR is one task of a github issue, please add `Master Issue: #<xyz>` to link to the master issue.)*

Master Issue: #<xyz>

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

Check the box below.

Need to update docs? 

- [x] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [ ] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

**Comment by tuteng** [comment]
https://kindly-ceramic-4ae.notion.site/Enable-jwt-authentication-cee2a6a1e24c4fcbbf56901d4c2ca47d cc @Huanli-Meng

File: charts/sn-platform/templates/streamnative-console/streamnative-console-statefulset.yaml [hunk 1]
```diff
@@ -177,7 +177,51 @@ spec:
           - name: GRAFANA_SERVLET_PATH
             value: {{ .Values.streamnative_console.configData.GRAFANA_SERVLET_PATH | default "/grafana/*" }}
           {{- end }}
-          {{- if or (eq .Values.auth.vault.enabled true) (eq .Values.streamnative_console.login.sso.google.enabled true) (eq .Values.streamnative_console.login.sso.azure.enabled true) (eq .Values.streamnative_console.login.sso.okta.enabled true) }}
+          {{- if .Values.streamnative_console.configData.INIT_DEFAULT_ENVIRONMENT }}
+          - name: DEFAULT_ENVIRONMENT_NAME
+            value: "pulsar"
+          - name: DEFAULT_ENVIRONMENT_SERVICE_URL
+            value: "http://{{ .Release.Name }}-sn-platform-broker:8080"
```
Comment (ciiiii) [inline charts/sn-platform/templates/streamnative-console/streamnative-console-statefulset.yaml:184]:
```suggestion
            value: "http://{{ template "pulsar.fullname" . }}-broker:8080"
```

**Review by ciiiii — COMMENTED** [review]
1. `.Values.components.vault: false` is required too.
   > https://kindly-ceramic-4ae.notion.site/Enable-jwt-authentication-cee2a6a1e24c4fcbbf56901d4c2ca47d
2. no jwt configuration for sla-detector 
    ```
    2022/11/09 06:17:51.060 log.go:50: [info] Start Pulsar detector with addr: :9000, service-url: pulsar://jwt-broker.jwt-auth.svc.cluster.local:6650, interval: 15s
    2022/11/09 06:17:51.060 detector_tasks.go:94: [fatal] missing configuration for token auth
    ```

File: charts/sn-platform/templates/streamnative-console/streamnative-console-statefulset.yaml (hunk 1 above)
Comment (tuteng) [reply charts/sn-platform/templates/streamnative-console/streamnative-console-statefulset.yaml]: Fixed, thanks

**Review by maxsxu — DISMISSED** [review]
FYI, the related doc: https://github.com/streamnative/charts/blob/master/docs/pulsar/how-to-enable-authentication-and-authorization.md

**Review by ciiiii — COMMENTED** [review]
Should we also clear unused legacy templates like this?
https://github.com/streamnative/charts/blob/9097c8d7e791a911863f401f75319a5c1317ebe1/charts/sn-platform/templates/proxy/_proxy.tpl#L18-L68

File: charts/sn-platform/values.yaml [hunk 2]
```diff
@@ -370,15 +371,15 @@ auth:
     enabled: true
   superUsers:
     # broker to broker communication
-    broker: "broker-admin"
+    broker: "admin,broker-admin,admin-approle,pulsar-manager-admin"
```
[the remark this answers is not in the document]
Comment (tuteng) [reply charts/sn-platform/values.yaml:374]: Fixed, I updated it to `.Values.auth.superUsers.client`

**Comment by tuteng** [comment]
@ciiiii PTAL

**Review by ciiiii — APPROVED** [review]

**Pull request merged**